services/objectstorage/client.go

```diff
@@ -82,8 +93,57 @@ func NewClient(clientOptions clientoptions.TerraformClientOptions, config *filec
 	)
 
 	cfg.HTTPClient = &http.Client{Transport: shared.CreateTransport(clientOptions.SkipTLSVerify, certificateAuthData)}
+
 	return &Client{
-		client:     objstorage.NewAPIClient(cfg),
-		fileConfig: config,
+		client:       objstorage.NewAPIClient(cfg),
+		fileConfig:   config,
+		signer:       signer,
+		globalRegion: globalRegion,
+	}
+}
+
+var (
+	regionToURL = map[string]string{
+		"":             "https://s3.eu-central-3.ionoscloud.com",
+		"eu-central-3": "https://s3.eu-central-3.ionoscloud.com",
+		"eu-central-4": "https://s3.eu-central-4.ionoscloud.com",
+		"us-central-1": "https://s3.us-central-1.ionoscloud.com",
+	}
+	ValidS3Locations = []string{"eu-central-3", "eu-central-4", "us-central-1"}
+)
+
+func (c *Client) ChangeConfigURL(bucketRegion string) error {
+	var region string
+
+	if c.globalRegion != "" && c.globalRegion != bucketRegion {
+		return fmt.Errorf("global region: %v is different from the bucket region: %v, if you want to use the bucket region, please unset the global region", c.globalRegion, bucketRegion)
+	}
+	region = bucketRegion
+	if region != "" && !slices.Contains(ValidS3Locations, region) {
```

Comment:
will this be a problem if someone declares in the config file a custom location and sets a custom url for it?

Reply:
@cristiGuranIonos done in 81f2e753c7853ff8ddfc6fa7f7f7b44b734f6df1